packages/assets-controllers/src/selectors/stringify-balance.ts

```diff
@@ -46,3 +48,46 @@ export function stringifyBalanceWithDecimals(
   }
   return `${whole}.${fractional}`;
 }
+
+/**
+ * Converts a decimal string representation back to a bigint balance.
+ * This is the inverse operation of stringifyBalanceWithDecimals.
+ *
+ * @param balanceString - The decimal string representation (e.g., "123.456")
+ * @param decimals - The number of decimals to apply (shifts decimal point right)
+ * @returns The balance as a bigint in the smallest unit
+ *
+ * @example
+ * parseBalanceWithDecimals("123.456", 18) // Returns 123456000000000000000n
+ * parseBalanceWithDecimals("0.001", 18)   // Returns 1000000000000000n
+ * parseBalanceWithDecimals("123", 18)     // Returns 123000000000000000000n
```

Comment:
Is the example up-to-date? It looks like it now returns a Hex val?

Reply:
Ah, thanks for raising. I started it returning a bigint, but it was giving me a lot of hassle with tests (you cannot run `toStrictEqual` with bigints because they cannot be serialised and deserialised to/from json), so I went for the less complicated Hex.

Will correct this.